<launch>

<!-- Gazebo Start -->
  <arg name="world" default="$(find DJi_M300_gazebo)/worlds/classroom.world" />
  <arg name="gui" default="true"/>
  <arg name="args" default=""/>
  
  <param name="/use_sim_time" value="true" />

  <node name="gazebo" pkg="gazebo_ros" type="gzserver" args="$(arg world) $(arg args)" respawn="false" output="screen"/>
  
    <!-- start gui -->
  <group if="$(arg gui)">
    <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen"/>
  </group>


<!-- Spawn Model-->
  <arg name="name" default="DJi_M300"/>
  <arg name="model" default="$(find DJi_M300_gazebo)/urdf/hku_m100.urdf.xacro"/>

  <arg name="x" default="0"/>
  <arg name="y" default="0"/>
  <arg name="z" default="0"/>
  <arg name="R" default="0"/>
  <arg name="P" default="0"/>
  <arg name="Y" default="0"/>

  <arg name="world_frame" default="world"/> <!-- This should actually be "/world". See https://github.com/ros-simulation/gazebo_ros_pkgs/pull/324 -->
  <arg name="base_link_frame" default="/base_link"/>

  <!-- send the robot XML to param server -->
  <param name="robot_description" command="$(find xacro)/xacro '$(arg model)' base_link_frame:=$(arg base_link_frame) world_frame:=$(arg world_frame)" />
  <param name="base_link_frame" type="string" value="$(arg base_link_frame)"/>
  <param name="world_frame" type="string" value="$(arg world_frame)"/>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="spawn_robot" pkg="gazebo_ros" type="spawn_model" output="screen"
        args="-param robot_description
           -urdf
           -x $(arg x)
           -y $(arg y)
           -z $(arg z)
           -model $(arg name)           
           -R $(arg R)
           -P $(arg P)
           -Y $(arg Y)"/>

  <!-- start robot state publisher -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen" >
    <param name="publish_frequency" type="double" value="50.0" />
  </node>

  <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher" />



<!-- Node Start 


<node pkg="DJi_M300_gazebo" type="DJi_M300_gazebo_bridge" name="DJi_M300_gazebo_bridge"/>

-->
 <node name="rviz" pkg="rviz" type="rviz" args="-d $(find DJi_M300_gazebo)/config/gazebo.rviz" />

</launch>
